### PR TITLE
API features

### PR DESCRIPTION
**Pull request by skynet67**
This branch has next API calls: 
- Users (get, update, delete)
- CreditCard (create, update, get, delete, acceptTerms, declineTerms, makeDefault, deactivate, reactivate)
- CreditCard verifications (select, verify)
- Device (create, update, get, delete)
- adpuPackages (get)

ECCKeys is internall class and all key exchange functionality was hidden from third party user

### CODE REVIEW

File: fitpay/src/main/java/com/fitpay/android/api/models/CreditCard.java [hunk 1]
```diff
@@ -211,61 +135,130 @@ public String getEncryptedData() {
...
-            return cvv;
-        }
+        /**
+         * Set credit card cvv2 code
+         * @param cvv cards's cvv2 code. string with 3 digits only
+         * @return a reference to this {@code Builder} object to fulfill the "Builder" pattern
+         */
+        public Builder setCVV(@NonNull String cvv) throws IllegalFormatException{
+
+            String pattern = "\\d{1,3}$";
```
Comment (ssteveli) [inline fitpay/src/main/java/com/fitpay/android/api/models/CreditCard.java]: I'm not sure how I feel about validation here.   By placing validation here, yes we gain a performance increase, however we sacrifice scalability.   We now need to release a new SDK if these rules ever change.   Let's just remove this and the ones below and let our API reject the request if the value is bad.

File: fitpay/src/main/java/com/fitpay/android/api/models/CreditCard.java [hunk 2]
```diff
@@ -211,61 +135,130 @@ public String getEncryptedData() {
...
-        public Address getAddress() {
-            return address;
+        /**
+         * Set credit card primary account number (PAN)
+         * @param pan cards's PAN. string with 16 digits only
+         * @return a reference to this {@code Builder} object to fulfill the "Builder" pattern
+         */
+        public Builder setPAN(@NonNull String pan) throws IllegalFormatException{
+
+            String pattern = "\\d{1,16}$";
```
Comment (ssteveli) [inline fitpay/src/main/java/com/fitpay/android/api/models/CreditCard.java]: Same as above.

File: fitpay/src/main/java/com/fitpay/android/utils/ApiManager.java [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+     * @param callback result callback
+     */
+    public void loginUser(String login, String password, final ApiCallback<Void> callback) {
+
+        String apiUrl = Constants.BASE_URL;
+        String data = String.format("{\"username\":\"%s\",\"password\":\"%s\"}", login, password);
+
+        Map<String, String> loginMap = new HashMap<>();
+        loginMap.put("response_type", "token");
+        loginMap.put("client_id", "pagare");
```
Comment (ssteveli) [inline fitpay/src/main/java/com/fitpay/android/utils/ApiManager.java]: This shouldn't be hard coded, the integrator will be provided a client_id to pass into the SDK at runtime.

File: fitpay/src/main/java/com/fitpay/android/utils/ApiManager.java [hunk 4]
```diff
@@ -0,0 +1,785 @@
...
+        loginMap.put("redirect_uri", apiUrl.substring(0, apiUrl.length() - 1));
```
Comment (ssteveli) [inline fitpay/src/main/java/com/fitpay/android/utils/ApiManager.java]: Same as above, the integrator will need to decide what the redirect_uri should be.

File: fitpay/src/main/java/com/fitpay/android/utils/ApiManager.java (hunk 3 above)
Comment (skynet67) [inline fitpay/src/main/java/com/fitpay/android/utils/ApiManager.java]: Now login call uses UserIndentity object as an input paramenter. So user should set username, password, clientId and redirectUri into this ojbect.

**Pull request merged**